### ISSUE TITLE
Adding extra files for directories in packaging process

### ISSUE THREAD

**Issue by alirezaghavaminia** [easybuilders/easybuild#639]
Hello, 
I have a suggestion. It would be nice if Easybuild user can specify files or directories in addition to the software files and directories to be included in the `package` file. 

For example, job files are some essential parts of an HPC ecosystem; it would be nice to be able to include them in the package file (e.g., .rpm or .deb file). The implementation of it must not be very hard since adding modules is already implemented. 

I would like to help in implementation if you tell me about it.

Thanks for your time.

**Comment by ocaisa** [comment]
Hi @alirezaghavaminia 

It's not really clear what you might mean by this. The job files example you give would be completely site specific (there are many schedulers out there) so is not something that connect with an easyconfig that we would distribute centrally. Can you give a more explicit example?

If you want to something specific to your site then I think creating your own [EasyBuild hook](https://easybuild.readthedocs.io/en/latest/Hooks.html) would be the right thing to do. If you specifically care about the packaging step you can hook directly into that using a `pre_package_hook`

**Comment by alirezaghavaminia** [comment]
Yes, you are right. The thing I had in mind was to add an option (via command line for example) to add extra files or directories to the packaging process. For example, I have already written some Jobfile templates which I want to be installed on a specific path during the installation of the main package, it would be nice to have something like `eb myprogram.eb --add-to-package=/path/to/file/now:/path/to/file/in/the/package` which at the process of making packages will add one more file to the generated package with the installation location of `/path/to/file/in/the/package`

Thanks for link to hook documentation. Probably it would do what I have in mind.
